apps/console/src/app.tsx

```diff
@@ -471,6 +487,7 @@ export const App: FunctionComponent<Record<string, never>> = (): ReactElement =>
                                             }
                                         />
                                         <ExtensionTemplatesProvider
+                                            shouldFetch={ hasApplicationTemplateViewPermissions }
```

Comment:
Hope this works for both root orgs and sub orgs. The last time I tried with the exact change, this check didn't work for sub orgs i.e templates were not fetched for sub orgs even if the user had scopes.

Reply:
Thank you for pointing this out. The current implementation does not work with sub-organizations when using the `ExtensionTemplateProvider` for application templates within the `app.tsx` file. To address this, I have refactored the code by removing the `ExtensionTemplateProvider` from the `app.tsx` file and adding it to the relevant features. While this results in the provider being repeated in both the connections and applications features, this is currently the most straightforward solution.

Addressed through https://github.com/wso2/identity-apps/pull/7237/commits/611f12513f08046611d72bc9357002bdda1f95de